components/raftstore-v2/src/operation/query/local.rs

```diff
@@ -184,17 +184,23 @@ where
             Ok(ReadRequestPolicy::StaleRead) => {
                 ReadResult::Ok((delegate, ReadRequestPolicy::StaleRead))
             }
-            // It can not handle other policies.
             // TODO: we should only abort when lease expires. For other cases we should retry
             // infinitely.
-            Ok(ReadRequestPolicy::ReadIndex) => ReadResult::Redirect,
+            Ok(ReadRequestPolicy::ReadIndex) => {
+                if req.get_header().get_replica_read() {
+                    ReadResult::Ok((delegate, ReadRequestPolicy::ReadIndex))
+                } else {
+                    ReadResult::Redirect
+                }
+            }
             Err(e) => ReadResult::Err(e),
         }
     }
 
     fn try_get_snapshot(
         &mut self,
         req: &RaftCmdRequest,
+        after_read_index: bool,
```

Comment:
This flag is quite odd.  It may be not needed if we choose to use Redirect to cover replica read.

Reply:
Redirect is not enough, because raftstore v2 no longer support getting snapshot. For replica read, we need to get a snapshot in local reader after a read index.